tycho-p2/tycho-p2-repository-plugin/src/main/java/org/eclipse/tycho/plugins/p2/repository/MavenP2SiteMojo.java

```diff
@@ -407,4 +414,77 @@ public String getClassifier() {
         return null;
     }
 
+    /**
+     *
+     * @return <code>true</code> if the pom configuration skip deploy for the project
+     */
+    protected boolean isSkippedDeploy(MavenProject mavenProject)
+    {
+        String property = mavenProject.getProperties().getProperty("maven.deploy.skip");
+        if (property != null) {
+            boolean skip = BooleanUtils.toBoolean(property);
+            getLog().debug("isSkippedDeploy " + mavenProject + " " + skip);
+            return skip;
+        }
+        String pluginId = "org.apache.maven.plugins:maven-deploy-plugin";
+        property = getPluginParameter(mavenProject, pluginId, "skip");
+        if (property != null) {
+            boolean skip = BooleanUtils.toBoolean(property);
+            getLog().debug("isSkippedDeploy " + mavenProject + " " + skip);
+            return skip;
+        }
+        if (mavenProject.getParent() != null) {
+            return isSkippedDeploy(mavenProject.getParent());
+        }
+        getLog().debug("isSkippedDeploy " + mavenProject + " " + false);
+        return false;
+    }
+
+    /**
+     * @param p        not null
+     * @param pluginId not null
+     * @param param    not null
+     * @return the simple parameter as String defined in the plugin configuration by <code>param</code> key
+     *         or <code>null</code> if not found.
+     * @since 2.6
+     */
+    private static String getPluginParameter(MavenProject p, String pluginId, String param)
```

Comment:
Maybe this is better placed in a common helper-class @mickaelistria any idea what one would be suitable?

Reply:
org.eclipse.tycho.core.utils.MavenSessionUtils might be a good place so this code could be reused by other mojos as well...